storage/innobase/handler/ha_innodb.cc

```diff
@@ -17080,6 +17080,52 @@ int innobase_rollback_by_xid(handlerton* hton, XID* xid)
 	}
 }
 
+/**
+This function is used to rollback one X/Open XA distributed transaction
+which is in the prepared state asynchronously.
+
+It only set the transaction's status to ACTIVE and persist the status.
+The transaction will be rolled back by background rollback thread.
+
+@param[in] hton InnoDB handlerton
+@param[in] xid X/Open XA transaction identification
+
+@return 0 or error number */
+int innobase_recover_rollback_by_xid(handlerton *hton, XID *xid)
+{
+	DBUG_ASSERT(hton == innodb_hton_ptr);
```

Comment:
I would not add an unused parameter `hton`, and I would pass `const XID &xid` because the parameter is not supposed to be `nullptr` nor to be modified by the callee.

Reply:
sure, addressed.